src/System.IO.Pipelines/tests/StreamPipeReaderTests.cs

```diff
@@ -519,7 +519,7 @@ public void NullStreamThrows()
         [Fact]
         public void InvalidBufferSizeThrows()
         {
-            Assert.Throws<ArgumentOutOfRangeException>(() => new StreamPipeReaderOptions(bufferSize: -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new StreamPipeReaderOptions(bufferSize: -2));
```

Comment:
It's funny that -1 this doesn't throw.

Reply:
Do you disagree with https://github.com/dotnet/corefx/issues/37984?  We can close this PR if the change isn't actually desirable.  I'm just trying to clear out the 3.0 backlog.

Reply:
The change was being taken so that we could change the buffer size in the future and have it impact existing apps if the current default shows it needs to be changed in the future.

We had the choice between 0 and -1 and went with -1 because that gives more flexibility around 0 in the future (it could be a real input that could represent something like "no buffering" rather than "default buffering").

Reply:
@tannergooding, why then did it focus only on the buffer sizes?  There are other arguments here that are hard coded numbers.

Reply:
Yea, also nullable seems like a more obvious way to indicate, lets use defaults.

Reply:
I'm going to close this PR and assign the issue back to you, Tanner.  This feels incomplete.

Reply:
As per the issue. Using `-1` to represent the "default" buffer size has prior art throughout the framework. The default buffer size is also something we have said we want to be able to freely change in the future.

There may be other bits that the same logic applies to but those should be taken independently as they may not have exact prior art. Its also been the case that the API surface has revved a few times since I initially opened the tracking bug (which was itself done as a side-effect of a previous API review), so anything "related" that wasn't covered may not have existed or may have been in a different location at the time.

Reply:
The discussion should be continued on the issue, not here.